src/main/java/com/github/deeepamin/ciaid/services/injectors/AbstractGitlabCIYamlScriptInjector.java

```diff
@@ -19,11 +20,24 @@
 
 import static com.github.deeepamin.ciaid.model.GitlabCIYamlKeywords.SCRIPT_KEYWORDS;
 
-public class GitlabCIYamlScriptInjector implements MultiHostInjector {
-  private static final Logger LOG = Logger.getInstance(GitlabCIYamlScriptInjector.class);
+/**
+ * Base class to allow injecting a shell script language into GitLab YAML elements.
+ * To avoid a compile-time dependency on the language, the ID is passed as a string.
+ */
+abstract class AbstractGitlabCIYamlScriptInjector implements MultiHostInjector {
+  private static final Logger LOG = Logger.getInstance(AbstractGitlabCIYamlScriptInjector.class);
+  private final @Nullable Language language;
```

Comment:
This is `@Nullable` to avoid breaking the class if anything unexpected happens, e.g. the language is unavailable.